src/main/java/com/hivemq/codec/decoder/mqtt3/Mqtt3SubackDecoder.java

```diff
@@ -65,9 +61,11 @@ public SUBACK decode(final Channel channel, final ByteBuf buf, final byte header
         while (buf.isReadable()) {
             final byte qos = buf.readByte();
             if (qos < 0 || qos > 2) {
-                log.error("A client (IP: {}) sent a suback which contained an invalid QoS. Disconnecting client.", getChannelIP(channel).or("UNKNOWN"));
-                eventLog.clientWasDisconnected(channel, "Invalid SUBACK qos");
-                channel.close();
+                disconnector.disconnect(channel,
+                        "A client (IP: {}) sent a SUBACK with an invalid QoS. Disconnecting client.",
+                        "Sent SUBACK with invalid QoS",
+                        Mqtt5DisconnectReasonCode.PROTOCOL_ERROR,
+                        null); // hivemq never sends SUBACKs reason string
```

Comment:
?

Reply:
actually i could not see the need for a reason string for a message HiveMQ never sends.

Reply:
the whole thing is pretty useless, as we use it only for testing.